### PR TITLE
Get rid of fxos-mvc dependency. r=gmarty

### PR DESCRIPTION
**Pull request by azasypkin**

### CODE REVIEW

File: app/js/controllers/base.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export default class BaseController {
```
Comment (azasypkin) [inline app/js/controllers/base.js]: I've migrated only things we actually use to have less technical debt and bias as possible (so no teardown and "view.init()" method).

File: app/js/controllers/main.js [hunk 2]
```diff
@@ -9,16 +8,23 @@ import DevController from './dev';
 
 import Foxbox from '../lib/foxbox/foxbox';
 
-export default class MainController extends RoutingController {
+const p = Object.freeze({
```
Comment (azasypkin) [inline app/js/controllers/main.js]: We used routing controller only in one place and we're going to migrate to some other framework, so here again I tried to migrate only things that we use and just merged RoutingController with MainController. If in the future we need it, it's pretty easy to extract it.

**Comment by azasypkin** [comment]
r? @gmarty 

As always current production version of this PR is deployed here: http://azasypkin.github.io/app

File: app/js/controllers/main.js (hunk 2 above)
Comment (gmarty) [inline app/js/controllers/main.js]: Have you considered moving the router to [react-router](https://github.com/reactjs/react-router)?
I think we should use it in the future, but this approach works for now.

Comment (azasypkin) [inline app/js/controllers/main.js]: > Have you considered moving the router to react-router?

Not yet

> I think we should use it in the future,

Yep, sounds good!

File: app/js/controllers/base.js (hunk 1 above)
Comment (gmarty) [inline app/js/controllers/base.js]: I like this.

**Comment by gmarty** [comment]
The patch looks good. I'm glad we're finally getting rid of this.
r+

**Comment by azasypkin** [comment]
Thanks for review!

**Pull request merged**